### PR TITLE
Prevent reacting to locked and archived threads

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/4568

## Description of Changes
Now preventing react to locked and archived threads

## Test Plan
- Lock a thread and try reacting to it -> it should be blocked
- Archive a thread and try reacting to it -> it should be blocked

## Deployment Plan
N/A

## Other Considerations
N/A

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx [hunk 1]
```diff
@@ -226,6 +228,7 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
             <ThreadCard
               key={thread.id + '-' + thread.readOnly}
               thread={thread}
+              canReact={!(!hasJoinedCommunity || !!thread.lockedAt || !!thread.archivedAt)}
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:231]: Can we make this logic more readable, maybe by assigning it to a variable first?

File: packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadOptions/ReactionButton/ReactionButton.tsx [hunk 2]
```diff
@@ -87,6 +88,7 @@ export const ReactionButton = ({
           onClick={async (e) => {
             e.stopPropagation();
             e.preventDefault();
+            if (disabled) return
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadOptions/ReactionButton/ReactionButton.tsx:91]: Missing semicolon?

**Review by jnaviask — CHANGES_REQUESTED** [review]
Doesn't seem to work? See: https://www.loom.com/share/4c46e37d4c764e15a6da9c2971f4eb13

To reproduce:
- give yourself admin in a community
- lock a thread
- try and react to it from the listing page -- see that your reaction appears to go through

File: packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx (hunk 1 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:231]: Done

File: packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadOptions/ReactionButton/ReactionButton.tsx [hunk 3]
```diff
@@ -52,6 +52,7 @@ export const ReactionButton = ({
   const handleSmallVoteClick = async (e) => {
     e.stopPropagation();
     e.preventDefault();
+    if (disabled) return
```
[the remark this answers is not in the document]
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadOptions/ReactionButton/ReactionButton.tsx:55]: Done

**Comment by mzparacha** [comment]
@jnaviask I think the thread you tried wasn't `locked`? Locked threads will have a lock icon
image

Maybe try with creating a thread -> locking it -> and then reacting to it.

**Review by jnaviask — APPROVED** [review]

**Pull request merged**